on gentoo with gcc 9.1.0, make compile not fail with -ftrack-macro-expansion=0
it works w/o this patch with =1 or =2

on archlinux, gcc 9.1.0 requires: -Wmisleading-indentation -Werror=misleading-indentation -ftrack-macro-expansion=0
to reproduce w/o patch!

libtool: compile:  x86_64-pc-linux-gnu-gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.22\" "-DPACKAGE_STRING=\"libblockdev 2.22\"" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.22\" "-DNDEBUG=/**/" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 "-DWITH_BD_BTRFS=/**/" "-DWITH_BD_DM=/**/" "-DWITH_BD_LOOP=/**/" "-DWITH_BD_MDRAID=/**/" "-DWITH_BD_SWAP=/**/" "-DWITH_BD_PART=/**/" "-DWITH_BD_FS=/**/" -DLIBMOUNT_NEW_ERR_API=1 -I. -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -Wall -Wextra -Werror -march=native -O2 -pipe -O2 -pipe -march=native -g0 -fno-var-tracking-assignments -fomit-frame-pointer -ftrack-macro-expansion=0 -std=gnu99 -c dbus.c  -fPIC -DPIC -o .libs/libbd_utils_la-dbus.o
In file included from /usr/include/glib-2.0/gio/gio.h:168,
                 from dbus.c:21:
/usr/include/glib-2.0/gio/glistmodel.h: In function 'G_IS_LIST_MODEL':
/usr/include/glib-2.0/gio/glistmodel.h:36:1: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
   36 | G_DECLARE_INTERFACE(GListModel, g_list_model, G, LIST_MODEL, GObject)
      | ^~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gio/glistmodel.h:36:1: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
/usr/include/glib-2.0/gio/glistmodel.h:36:1: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
/usr/include/glib-2.0/gio/glistmodel.h:36:1: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
In file included from /usr/include/glib-2.0/gio/gio.h:169,
                 from dbus.c:21:
/usr/include/glib-2.0/gio/gliststore.h: In function 'G_IS_LIST_STORE':
/usr/include/glib-2.0/gio/gliststore.h:36:1: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
   36 | G_DECLARE_FINAL_TYPE(GListStore, g_list_store, G, LIST_STORE, GObject)
      | ^~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/gio/gliststore.h:36:1: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
/usr/include/glib-2.0/gio/gliststore.h:36:1: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
/usr/include/glib-2.0/gio/gliststore.h:36:1: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
cc1: all warnings being treated as errors
make[2]: *** [Makefile:575: libbd_utils_la-dbus.lo] Error 1
make[2]: Leaving directory '/var/tmp/portage/sys-libs/libblockdev-2.22/work/libblockdev-2.22/src/utils'
make[1]: *** [Makefile:429: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/sys-libs/libblockdev-2.22/work/libblockdev-2.22/src'
make: *** [Makefile:508: all-recursive] Error 1
 * ERROR: sys-libs/libblockdev-2.22::gentoo failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=sys-libs/libblockdev-2.22::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=sys-libs/libblockdev-2.22::gentoo'`.
 * The complete build log is located at '/tmp/ebuild.logs/build/sys-libs/libblockdev-2.22:20190630-162209.log'.
 * For convenience, a symlink to the build log is located at '/var/tmp/portage/sys-libs/libblockdev-2.22/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sys-libs/libblockdev-2.22/temp/environment'.
 * Working directory: '/var/tmp/portage/sys-libs/libblockdev-2.22/work/libblockdev-2.22'
 * S: '/var/tmp/portage/sys-libs/libblockdev-2.22/work/libblockdev-2.22'

and

x86_64-pc-linux-gnu-gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.22\" "-DPACKAGE_STRING=\"libblockdev 2.22\"" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.22\" "-DNDEBUG=/**/" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 "-DWITH_BD_BTRFS=/**/" "-DWITH_BD_DM=/**/" "-DWITH_BD_LOOP=/**/" "-DWITH_BD_MDRAID=/**/" "-DWITH_BD_SWAP=/**/" "-DWITH_BD_PART=/**/" "-DWITH_BD_FS=/**/" -DLIBMOUNT_NEW_ERR_API=1 -I. -I./../../../include/ -I./../ -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/blkid -I/usr/include/libmount -I/usr/include/blkid -Wall -Wextra -Werror -march=native -O2 -pipe -Werror=misleading-indentation -ftrack-macro-expansion=0 -O2 -pipe -march=native -std=gnu99 -c generic.c  -fPIC -DPIC -o .libs/libbd_fs_la-generic.o
generic.c: In function 'get_fs_info':
generic.c:76:5: error: this 'else' clause does not guard... [-Werror=misleading-indentation]
   76 |     g_return_val_if_fail (type != NULL, NULL);
      |     ^~~~~~~~~~~~~~~~~~~~
generic.c:76:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'else'
cc1: all warnings being treated as errors

--- glib-2.58.3/gobject/gtype.h	2019-01-21 14:38:50.000000000 +0100
+++ fixed/gobject/gtype.h	2019-06-30 19:16:13.779332597 +0200
@@ -2283,12 +2283,12 @@ const gchar *    g_type_name_from_class
 #ifdef	__GNUC__
 #  define _G_TYPE_CIT(ip, gt)             (G_GNUC_EXTENSION ({ \
   GTypeInstance *__inst = (GTypeInstance*) ip; GType __t = gt; gboolean __r; \
-  if (!__inst) \
-    __r = FALSE; \
-  else if (__inst->g_class && __inst->g_class->g_type == __t) \
-    __r = TRUE; \
-  else \
-    __r = g_type_check_instance_is_a (__inst, __t); \
+  if (!__inst) { \
+    __r = FALSE; }\
+  else { if (__inst->g_class && __inst->g_class->g_type == __t) {\
+    __r = TRUE; }\
+  else {\
+    __r = g_type_check_instance_is_a (__inst, __t); }}\
   __r; \
 }))
 #  define _G_TYPE_CCT(cp, gt)             (G_GNUC_EXTENSION ({ \
--- glib-2.58.3/glib/gmacros.h	2019-01-21 14:38:50.000000000 +0100
+++ fixed/glib/gmacros.h	2019-06-30 21:41:14.544949160 +0200
@@ -415,11 +415,11 @@
 #define _G_BOOLEAN_EXPR(expr)                   \
  G_GNUC_EXTENSION ({                            \
    int _g_boolean_var_;                         \
-   if (expr)                                    \
+   if (expr) {                                   \
       _g_boolean_var_ = 1;                      \
-   else                                         \
+   } else {                                        \
       _g_boolean_var_ = 0;                      \
-   _g_boolean_var_;                             \
+   } _g_boolean_var_;                             \
 })
 #define G_LIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR((expr)), 1))
 #define G_UNLIKELY(expr) (__builtin_expect (_G_BOOLEAN_EXPR((expr)), 0))
